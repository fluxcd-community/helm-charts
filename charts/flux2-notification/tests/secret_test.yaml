suite: test secret deployment
templates:
  - secret.yaml
tests:
  - it: should be empty if secret.create is false
    asserts:
      - hasDocuments:
          count: 0
  - it: should have kind Secret for k8s >= 1.21
    capabilities:
      majorVersion: 1
      minorVersion: 21
    set:
      secretlist:
      - name: slack-token
        value:
          data:
            address: <BASE64>
            token: <BASE64>
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - isAPIVersion:
          of: v1
      - equal:
          path: metadata.name
          value: slack-token
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
    set:
      secretlist:
      - name: webhook-url
        value:
          stringData:
            headers: |
              Authorization: token
              X-Forwarded-Proto: https
