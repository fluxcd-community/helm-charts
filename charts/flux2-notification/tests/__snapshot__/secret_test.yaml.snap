should match snapshot of default values:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        app.kubernetes.io/version: 0.31.3
        helm.sh/chart: flux2-notification-1.0.0
      name: webhook-url
      namespace: NAMESPACE
    stringData:
      headers: |
        Authorization: token
        X-Forwarded-Proto: https
    type: Opaque
